scio-avro/src/main/scala/com/spotify/scio/coders/AvroBytesUtil.scala

```diff
@@ -17,41 +17,47 @@
 
 package com.spotify.scio.coders
 
-import org.apache.avro.generic.{GenericData, GenericRecord}
-import org.apache.avro.{Schema => ASchema}
+import org.apache.avro.generic.{GenericDatumWriter, GenericRecord, GenericRecordBuilder}
+import org.apache.avro.io.{DatumWriter, Encoder}
+import org.apache.avro.{Schema, SchemaBuilder}
 import org.apache.beam.sdk.coders.{Coder => BCoder}
+import org.apache.beam.sdk.extensions.avro.io.AvroSink.DatumWriterFactory
 import org.apache.beam.sdk.util.CoderUtils
 
 import java.nio.ByteBuffer
-import scala.jdk.CollectionConverters._
 
 private[scio] object AvroBytesUtil {
-  val schema: ASchema = {
-    val s = ASchema.createRecord("AvroBytesRecord", null, null, false)
-    s.setFields(
-      List(
-        new ASchema.Field(
-          "bytes",
-          ASchema.create(ASchema.Type.BYTES),
-          null,
-          null.asInstanceOf[Object]
-        )
-      ).asJava
-    )
-    s
+  val schema: Schema = SchemaBuilder
+    .record("AvroBytesRecord")
+    .fields()
+    .requiredBytes("bytes")
+    .endRecord()
+
+  private val byteField = schema.getField("bytes")
+
+  def datumWriterFactory[T: Coder]: DatumWriterFactory[T] = {
+    val bCoder = CoderMaterializer.beamWithDefault(Coder[T])
```

Comment:
Logic change here since we do not use the coder options for byte serialization. I think this cased of an issue when consuming data from PubSub, and tunring one Nullable, records could not be read.